pwiz_tools/Skyline/FileUI/ExportMethodDlg.cs

```diff
@@ -902,13 +903,38 @@ public void OkDialog(string outputPath)
 
             if (Equals(InstrumentType, ExportInstrumentType.BRUKER_TIMSTOF))
             {
-                var missingIonMobility = BrukerTimsTofIsolationListExporter.GetMissingIonMobility(documentExport, _exportProperties);
-                if (missingIonMobility.Length > 0)
+                if (!BrukerTimsTofIsolationListExporter.CheckIonMobilities(documentExport, _exportProperties,
+                        templateName, out var missing, out var outOfRange, out var limitLower, out var limitUpper))
                 {
-                    MessageDlg.Show(this,
-                        Resources.ExportMethodDlg_OkDialog_All_targets_must_have_an_ion_mobility_value__These_can_be_set_explicitly_or_contained_in_an_ion_mobility_library_or_spectral_library__The_following_ion_mobility_values_are_missing_ +
-                        Environment.NewLine + Environment.NewLine +
-                        TextUtil.LineSeparate(missingIonMobility.Select(k => k.ToString())));
+                    var message = new StringBuilder();

```

Comment:
Whenever I am building up a big message like this, I usually use a List&lt;string> for the lines. Then, I call "TextUtil.LineSeparate" at the end of it all. One thing that that makes easier is not having to worry about there being an extra line break at the end of it all.